spring-session/src/main/java/org/springframework/session/data/hazelcast/config/annotation/web/http/EnableHazelcastHttpSession.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2002-2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.session.data.hazelcast.config.annotation.web.http;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.session.config.annotation.web.http.EnableSpringHttpSession;
+
+/**
+ * Add this annotation to a {@code @Configuration} class to expose the
+ * SessionRepositoryFilter as a bean named "springSessionRepositoryFilter" and
+ * backed by Hazelcast. In order to leverage the annotation, a single {@link HazelcastInstance}
+ * must be provided. For example:
+ * <pre>
+ * <code>
+ * {@literal @Configuration}
+ * {@literal @EnableHazelcastHttpSession}
+ * public class HazelcastHttpSessionConfig {
+ *
+ *     {@literal @Bean}
+ *     {@literal @ConditionalOnMissingBean}
+ *     public HazelcastInstance hazelcastInstance(Config hazelcastConfig) {
+ *         if (StringUtils.hasText(hazelcastConfig.getInstanceName())) {
+ *             return Hazelcast.getOrCreateHazelcastInstance(hazelcastConfig);
+ *         }
+ *         return Hazelcast.newHazelcastInstance(hazelcastConfig);
+ *     }
+ *
+ * }
+ * </code>
+ * </pre>
+ *
+ * More advanced configurations can extend {@link HazelcastHttpSessionConfiguration} instead.
+ * 
+ * @author Tommy Ludwig
+ * @since 1.1
+ * @see EnableSpringHttpSession
+ */
+@Retention(value=java.lang.annotation.RetentionPolicy.RUNTIME)
+@Target(value={java.lang.annotation.ElementType.TYPE})
+@Documented
+@Import(HazelcastHttpSessionConfiguration.class)
+@Configuration
+public @interface EnableHazelcastHttpSession {
+	int maxInactiveIntervalInSeconds() default 1800;
```

Comment:
I wanted to maintain the functionality in a very similar way to the Redis equivalent in order to lower the barrier to entry. You are right that Spring Session will always overwrite this value, which can be confusing if you configure the same thing externally. However, Spring Session would only be configuring the specific `Map` used for the `SessionRepository` which seems like it might acceptable and the same as the Redis configuration does, from what I can tell.

If we do want to avoid overwriting, I think we would have to change to an `Integer` and make the default for this `null` (I think -1 would be potentially confusing and could be mistaken for a valid value when it is not). Then only overwrite the value (Hazelcast's default `maxIdleSeconds` is 0) if this is not null.
What do you think?